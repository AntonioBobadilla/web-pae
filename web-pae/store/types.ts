import { Subject } from '@/components/search-bar';

export const days: {
  [key: string]: number;
} = {
  lunes: 0,
  martes: 1,
  miercoles: 2,
  jueves: 3,
  viernes: 4,
  sabado: 5
};

export interface Period {
  dia: string;
  inicio: number;
  fin: number;
  id: string;

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
}

// Define a type for the slice state
export interface CreateTutorState {
  name: string;
  email: string;
  major: string;
  password: string;
  passwordConfirmation: string;
  schedule: {
    firstPeriod: Period[];
    secondPeriod: Period[];
    thirdPeriod: Period[];
  };
  subjects: Subject[];
  isLoading: boolean;
  error: string;
}
