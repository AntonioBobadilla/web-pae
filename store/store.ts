import { configureStore } from '@reduxjs/toolkit';
import { createWrapper } from 'next-redux-wrapper';
import { Action } from 'redux';
import thunk, { ThunkAction } from 'redux-thunk';
import createTutor from './reducers/create-tutor';
import scheduleTutoring from './reducers/schedule-tutoring';
import user from './reducers/user';

const saveToLocalStorage = (state: RootState) => {
  try {
    localStorage.setItem('state', JSON.stringify(state));
  } catch (e) {
    // console.error(e);
  }
};

const loadFromLocalStorage = () => {
  try {
    const stateStr = localStorage.getItem('state');
    return stateStr ? JSON.parse(stateStr) : undefined;
  } catch (e) {
    // console.error(e);
    return undefined;
  }
};

const persistedStore = loadFromLocalStorage();

// middleware
const middleware = [thunk];

// creating store

export const store = configureStore({
  reducer: {
    user,
    scheduleTutoring,
    createTutor
  },
  middleware,
  preloadedState: persistedStore
});

store.subscribe(() => {
  saveToLocalStorage(store.getState());
});

// assigning store to next wrapper
export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

const makeStore = () => store;

export const wrapper = createWrapper(makeStore);
