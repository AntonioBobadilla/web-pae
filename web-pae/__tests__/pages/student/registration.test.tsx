import RegisterForm from '@/components/register-form';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { store } from 'store/store';

describe('Student registration', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <RegisterForm
          nextStep={() => console.log('ok')}
          student
          isLoading={false}
        />
      </Provider>
    );
  });
  // TODO: fix this test
  // it('renders tutor registration page unchanged', () => {
  //   const { container } = render(<Registration />);
  //   expect(container).toMatchSnapshot();
  // });
  it('should be disabled', () => {
    expect(screen.getByRole('button')).toBeEnabled();
  });
});
