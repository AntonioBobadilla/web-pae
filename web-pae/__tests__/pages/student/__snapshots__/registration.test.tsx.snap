// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student registration renders admin profile page unchanged 1`] = `
<div>
  <h1>
    PROFILE
  </h1>
</div>
`;

exports[`Student registration renders student registration page unchanged 1`] = `
<div>
  <h1>
    PROFILE
  </h1>
</div>
`;

exports[`Student registration renders student-registration page unchanged 1`] = `
<div>
  <div
    class="container containerPopup"
  >
    <h1
      class="title"
    >
      REGISTRO
    </h1>
    <h2
      class="subtitle"
    >
      Datos personales
    </h2>
    <form
      class="registerForm"
    >
      <div
        class="input"
      >
        <input
          class="textInput"
          id="name"
          name="name"
          placeholder="NOMBRE COMPLETO*"
          value=""
        />
      </div>
      <div
        class="input"
      >
        <input
          class="textInput"
          id="email"
          name="email"
          placeholder="CORREO INSTITUCIONAL*"
          value=""
        />
      </div>
      <div
        class="input"
      >
        <input
          class="textInput"
          id="password"
          name="password"
          placeholder="CONTRASEÑA*"
          type="password"
          value=""
        />
      </div>
      <div
        class="input"
      >
        <input
          class="textInput"
          id="passwordConfirmation"
          name="passwordConfirmation"
          placeholder="CONFIRMAR CONTRASEÑA*"
          type="password"
          value=""
        />
      </div>
      <div
        class="button"
      >
        <button
          class="button primary"
          disabled=""
          type="submit"
        >
          CONCLUIR REGISTRO
        </button>
      </div>
    </form>
  </div>
</div>
`;
